pymongo/common.py

```diff
@@ -817,9 +821,13 @@ def validate_auth_option(option: str, value: Any) -> tuple[str, Any]:
 def validate(option: str, value: Any) -> tuple[str, Any]:
     """Generic validation function."""
     lower = option.lower()
-    validator = VALIDATORS.get(lower, raise_config_error)
-    value = validator(option, value)
-    return option, value
+    try:
+        validator = VALIDATORS[lower]
+        value = validator(option, value)
```

Comment:
Let's move the validator() call to after the try/except so that we only ever handle KeyErrors from the dict look up.